### DISCUSSION TITLE
Ping slot channel frequency for CN470 implemented according to LoRaWAN specification v1.0.2

### DISCUSSION THREAD

**Discussion by lincy13** [Lora-net/LoRaMac-node#1378] [Q&A] [answered]
In tag v4.4.7 or other CN470 implemented according to LoRaWAN specification v1.0.2, 
`#define CN470_PING_SLOT_CHANNEL_FREQ                508300000`
But the regional parameter revision does not specify the rules for CN470 ping slot frequency, 
which leads to a problem, the definition of ping slot frequency in the server is different from here.
Who should I do and implement compatible processing?

**Comment by mluis1** [comment, marked as answer]
The tag v4.4.7 is based on LoRaWAN 1.0.3 and LoRaWAN Regional Parameters 1.0.3reva specifications.

The ping slot frequencies are a function of the beacon frequency which hops. The LoRaWAN specification says the following for the ping slots frequency

![image](https://user-images.githubusercontent.com/5763295/200502022-ff1d9b10-c712-4777-b7e6-a703167d1bcc.png)

The code provided in this project follows the specifications. If the used server does not follow the specified rules I would recommend to contact your network server provider.

Please note as well that v4.4.7 tag is quite old now. One should upgrade the code to the latest versions of this repository as a lot of things have evolved since then.
